### DISCUSSION TITLE
Advice for sudden unstable network

### DISCUSSION THREAD

**Discussion by majorsl** [Koenkk/zigbee2mqtt#14467] [General]
I moved over to z2m about a month ago from deconz.  I needed support for more devices than it offered.  Instead of using the Conbee II, I went with the popular Sonoff Zigbee 3.0 USB Dongle Plus (I believe the "P" version and not the newer and slightly smaller v2 "E" version). I used whatever the recommended coordinator firmware was from the main page as of 30 days ago. I run z2m in Docker on an RPi.  My zigbee channel is far away from my 802.11 channel (spectrum wise, opposite ends). We're in the country so no neighbors with conflicting wifi.

I built out my network by adding powered (routing) devices first from the controller out, and then wireless ones.  Everything was pure awesome, stable and fast.

When 1.28.0 came along, I figured that was a good time to shutdown the RPi and make a backup of the whole SD card.  Since everything was down for about 30 minutes, I'm sure the Zigbee network tried to do re-paths in an effort to get to the coordinator.  When it came back on-line, everything showed up, but since then I've had random powered and battery devices just go Offline. Repairing them solves the problem, but it continues to happen with different devices.

I have 81 devices, 47 of them are router devices so my mesh network is pretty dispersed so I don't understand why devices are going offline as there should be many stable routes to the coordinator.  Some of the powered devices that randomly went off-line are smack in the middle of the mesh with LQIs in the 185-225 range.

**Comment by desertblade** [comment]
After months of a stable Z2M, I was having some stability issues. I also had a bunch of powered repeaters.  Discovered that my WIFI network on my main floor was using channel 1, which is nearly the same frequency as the default Zigbee channel 11.  

Moving my WIFI away from Channel 1 improved performance. I did not change anything recently.  My only guess is maybe my neighbor changed their router, since it mostly affected devices on one side of the house. Something probably increased the noise floor and added additional interference.

I tried using the Availability setting and I found no matter what I did Z2M was always kicking devices and they never came back. Now I keep the setting to "Availability Simple".

**Reply by majorsl** [reply]
I'm fairly sure Wifi isn't it.  Zigbee is channel 11 and Wifi is channel 11, but that's opposite ends of the spectrum.  I have no close neighbors and I installed a Wifi scanner on my phone and walked around, nothing on 802.11 but mine.

I'm beginning to suspect one or more routers are having issues.  I have had two wall switches responding to commands from z2m but when triggered at the wall switch, they show no state change even in the logs. It's like the "send" is never making it.  I re-paired them directly to the Coordinator and they work perfectly.  I'm leaning towards a couple of cheap bulbs that are also routers as the culprit.

I will try the "Availability Simple" setting, I was using Advanced to reduce the chatter in the log by extending the check times, but I think something else is still going on with my network routing.

Edit: Oh, and I've also seen some 3-5 second delays in toggling devices, again, they appear to be close to those bulbs.

**Reply by davew21** [reply]
Load up your map and have a look see if something is constantly sending, I had an IKEA ICTC-G-1 dimmer that would randomly start sending all the time even though the lamp it was connected to didn’t show any issues and it caused all sorts of strange issues including delays, as soon as I took the battery out of it everything went back to normal.

**Reply by majorsl** [reply]
> Load up your map and have a look see if something is constantly sending, I had an IKEA ICTC-G-1 dimmer that would randomly start sending all the time even though the lamp it was connected to didn’t show any issues and it caused all sorts of strange issues including delays, as soon as I took the battery out of it everything went back to normal.

Thanks! I clicked "Map" and of course see all my devices and their links, but I don't see how to show send (or receive) of devices. I only see LQI values.

**Reply by davew21** [reply]
Each device on your map has a white ring around it, every time the device sends that ring flashes green, if you have one constantly flashing green that could be your problem, it should also show up in the log as well

**Reply by majorsl** [reply]
> Each device on your map has a white ring around it, every time the device sends that ring flashes green, if you have one constantly flashing green that could be your problem, it should also show up in the log as well

Okay, no devices are very chatty at all, even in the logs, it is what I expect.

I clearly have routing issues going on, but I'm at a loss to know how to solve them.  For example, I have two switches in the kitchen that will go off-line, but with good quality links. I discovered that while they are showing Offline, when I toggle the GROUP they are in (Kitchen lights) they both respond along with the others in that group.

They are still joined to the network, but z2m isn't able to speak to them directly for some reason.

**Reply by vschild** [reply]
I moved my network to channel 25 2 months ago and it was working perfectly but then this week somo devices started to disconnect. I live outside the city so interference is not a big issue.

**Comment by majorsl** [comment]
Things are more stable now. I noticed in Home Assistant that two GE 45856GE wall switches were off-line multiple times over a 24 hour period, even tho they had great LQI. These are Jasco/Enbrighten, but are older ones and I don't think are Zigbee 3.0. The Zwave version of them from about the same time are known to have bad capacitors.

I also have always suspected my cheap eWeLink Smart Plugs too.  I removed them from my network.

Everything has been stable for 10 days now.  I don't have a use for those Smart Plugs now, but if I do in the future, I may add them back and see if things fall apart again.  It'd be nice to know if it was just the wall switches.

I thought if you had a bad device or one that went off-line, the point of the mesh was to self-heal.  I guess, not-so-much.
